src/main/controllers/CrownFirmListController.ts

```diff
@@ -0,0 +1,46 @@
+import {Response} from 'express';
+import {PipRequest} from '../models/request/PipRequest';
+import {cloneDeep} from 'lodash';
+import moment from 'moment';
+import { PublicationService } from '../service/publicationService';
+import { LocationService } from '../service/locationService';
+import { DataManipulationService } from '../service/dataManipulationService';
+import {CrownFirmListService} from '../service/listManipulation/crownFirmListService';
+
+const publicationService = new PublicationService();
+const locationService = new LocationService();
+const dataManipulationService = new DataManipulationService();
+const firmListService = new CrownFirmListService();
+
+export default class CrownFirmListController {
```

Comment:
THink we are missing the unit test for the controller. Could you add that in please? :)

Reply:
now done :)